### DISCUSSION TITLE
[Feature] Allow a systemwide Scoop install that still works within each individual user folder

### DISCUSSION THREAD

**Discussion by fmillion** [ScoopInstaller/Scoop#5374] [Help and Questions]
#### Is your feature request related to a problem? Please describe.
I want to be able to have the `scoop` command available to all users on a system. However, I want the command to work for each user as if it were installed by that user locally - i.e. program installs go to each user's own home directory.

I basically want to be able to preinstall Scoop on a machine, and then anyone who logs into that machine has access to use Scoop to install programs within their own user account without needing to install it themselves. 

#### Describe the solution you'd like

Perhaps it already exists, but I haven't found anything obvious in the documentation for it.

Perhaps something like:

    # I think you'd have to do this so that all users have access to powershell?
    Set-ExecutionPolicy RemoteSigned -Scope LocalMachine

    # install scoop to an admin account on the machine
    irm get.scoop.sh | iex

    # after scoop installed into an administrator account, also install it for everyone
    scoop install --global scoop

    # and now any user, upon logging in, can simply open powershell and start using scoop to install stuff.
    # even non-admin users, since it will just install to their local user account anyway.
    # Each user might want to start with scoop update *
    # if that could be scripted to happen the first time a user runs scoop, even better.

While for *my* use case being able to limit access to Scoop by user/group is unimportant, I'd imagine this would be important for some, so perhaps a consideration for how that would be handled. 

#### Describe alternatives you've considered

I could maybe set up some sort of login script on a local machine for the DefaultUser account, but I'm a bit rusty on my Windows admin skills.

**Comment by rashil2000** [comment]
I'm just trying to understand, what problem are you facing when installing Scoop for each user individually (i.e. in their respective home folders)?

**Comment by fmillion** [comment]
I don't want to have to log in to someone's account to install scoop for them. 

For example, in an educational setting I want to prepare a few machines for use by multiple student users. I want each account created to be able to have access to Scoop automatically.

Another example: on a domain-joined computer, I want developers to have access to scoop on any computer they happen to log in to, without having to install it manually each time.

Sure, it might be easy to install it for the end user, but the convenience of it just being available automatically is a nice quality-of-life feature.

I still use Scoop because it has plenty of apps that WinGet doesn't (and may never, with its limitations) have. But WinGet is built in to Windows and requires no installation. I want a similar seamless experience for users for Scoop on systems I admin.

**Comment by rashil2000** [comment]
> For example, in an educational setting I want to prepare a few machines for use by multiple student users. I want each account created to be able to have access to Scoop automatically.

I don't think this is possible, given the current architecture of Scoop. In fact, I'm not aware of any package manager (even on Linux/macOS) that allows this. You want a global _package manager_, but local _packages_. Even if winget is a global package manager, the packages it installs are also global.

Here's an idea you could try: install Scoop on a global location, but then set the SCOOP environment variable for each user pointing to their respective home folders (more info on env vars here: https://github.com/ScoopInstaller/Install/).

**Reply by fmillion** [reply]
I'll try the environment variable option, that might actually work.

Some package managers do work on a user level. Easiest example that comes to mind is pip (and similarly anaconda) for Python - if you run as a regular user, it'll install packages in the user's home directory, but if run as root it'll install globally. You can install Python and Pip (or Anaconda) globally, but any user can run them to install packages just for their own user. That's kind of what I was aiming for with Scoop.

**Reply by itu-itis22-gokalpay22** [reply]
I would like to see this feature implemented too!
